### PR TITLE
Storage Initializer: Support virtual path style in S3

### LINKED ISSUE #2883: Support ali-cloud oss in storage-uri
/kind feature

**Describe the solution you'd like**
There is already a closed issue #2128  requesting support for Aliyun OSS links for storage-uri. The reason for closing this issue was that "OSS is compatible with the S3 API". But it's not true. Storage-initializer will throw a error msg "Please use virtual hosted style to access" if use OSS links as S3 links.
image

This may be due to AliCloud OSS not supporting path-style URLs. As it's mentioned in [Compatibility with Amazon S3](https://www.alibabacloud.com/help/en/object-storage-service/latest/s3-compatibility-overview#concept-2158780), "For  security reasons, OSS supports only virtual hosted-style URLs in requests". But  S3 support both path-style and virtual-hosted–style URLs. 

Is it possible to add support for AliCloud OSS URLs?

**Anything else you would like to add:**
It seems KServe provide an environment variable S3_USER_VIRTUAL_BUCKET to specify URL style，but not used in storage.py. This may be a work-around if S3_USER_VIRTUAL_BUCKET can work.
image
image

### PR DESCRIPTION
**Pull request by lizzzcai**
**What this PR does / why we need it**:

Support virtual path style in S3 so other S3 compatible storage like `Alibaba Cloud OSS` can be supported via `s3://`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2883 

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] Unit tests on the S3 config
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes. - No

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Storage initializer supports virtual path style in S3
```

### CODE REVIEW

File: python/kserve/kserve/storage/storage.py [hunk 1]
```diff
@@ -136,12 +136,21 @@ def _update_with_storage_spec():
...
     @staticmethod
     def get_S3_config():
+        # default s3 config
+        c = Config()
+
         # anon environment variable defined in s3_secret.go
         anon = ("True" == os.getenv("awsAnonymousCredential", "false").capitalize())
+        # S3UseVirtualBucket environment variable defined in s3_secret.go
+        # use virtual hosted-style URLs if enabled
+        virtual = ("True" == os.getenv("S3_USER_VIRTUAL_BUCKET", "false").capitalize())
```
Comment (yuzisun) [inline python/kserve/kserve/storage/storage.py]: We can use lower() to do case insensitive checks, with the above check if someone sets to TRUE then it won’t work.

Comment (lizzzcai) [reply python/kserve/kserve/storage/storage.py]: done

**Comment by yuzisun** [comment]
Thanks @lizzzcai !

**Pull request merged**